### PR TITLE
Detail List Subcomponent 

### PR DESCRIPTION
**Pull request by emilyrohrbough**
### Summary
Added the subcomponents Detail List and Detail List Item to the Detail View package. These components were designed to have the behavior of the float-grid in terra-view-models and is expected to be used as an item in the DetailView details prop. 

@cerner/terra

### CODE REVIEW

File: packages/terra-clinical-detail-view/src/DetailListItem.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.terraClinical-DetailListItem {
+  min-width: 50%;
+
+  padding-bottom: 0.714rem;
+  padding-right: 0.714rem;
```
Comment (dkasper-was-taken) [inline packages/terra-clinical-detail-view/src/DetailListItem.scss]: For padding left and right, you might consider the terra-padding mixin in the case of bidi.

File: packages/terra-clinical-detail-view/src/DetailListItem.jsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+};
+
+const DetailListItem = ({ item, ...customProps }) => {
+  const detailListItemClassNames = classNames([
+    'terraClinical-DetailListItem',
+    customProps.className,
+  ]);
+
+  return (
+    <div className={detailListItemClassNames}>
```
Comment (dkasper-was-taken) [inline packages/terra-clinical-detail-view/src/DetailListItem.jsx]: Does this element need customProps unpacked on it?

File: packages/terra-clinical-detail-view/src/DetailListItem.scss (hunk 1 above)
Comment (emilyrohrbough) [inline packages/terra-clinical-detail-view/src/DetailListItem.scss]: Added it. I did thinking about the bidi-mixin, but must not have used it.

File: packages/terra-clinical-detail-view/src/DetailListItem.jsx (hunk 2 above)
Comment (emilyrohrbough) [inline packages/terra-clinical-detail-view/src/DetailListItem.jsx]: Yes customProps, I added them the detail list and detail list items.

File: packages/terra-clinical-detail-view/src/DetailListItem.scss [hunk 3]
```diff
@@ -1,7 +1,8 @@
+@import '~terra-mixins';
+
 .terraClinical-DetailListItem {
   min-width: 50%;
 
-  padding-bottom: 0.714rem;
-  padding-right: 0.714rem;
+  @include terra-padding-horizontal (0.714rem, 0.714rem);
```
Comment (dkasper-was-taken) [inline packages/terra-clinical-detail-view/src/DetailListItem.scss]: Did you mean to remove padding on the bottom and add left padding?

Comment (emilyrohrbough) [inline packages/terra-clinical-detail-view/src/DetailListItem.scss]: Nope! [Fixed](https://github.com/cerner/terra-clinical/pull/16/commits/89b6ad42cac7e8cbfae1a5f5fff25019312215ea#diff-ab3e61aee5ded6313560c505766ff869R7)

File: packages/terra-clinical-detail-view/src/DetailListItem.scss [hunk 4]
```diff
@@ -0,0 +1,9 @@
+@import '~terra-mixins';
+
+.terraClinical-DetailListItem {
+  min-width: 50%;
+
+  @include terra-padding-end (0.714rem);
```
Comment (jmsv6d) [inline packages/terra-clinical-detail-view/src/DetailListItem.scss]: nit: remove space before parenthesis

File: packages/terra-clinical-detail-view/src/DetailList.jsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+import './DetailList.scss';
+import DetailListItem from './DetailListItem';
+
+const propTypes = {
+  /**
+   * The title to be displayed abo list.
```
Comment (StephenEsser) [inline packages/terra-clinical-detail-view/src/DetailList.jsx]: I think there are a few letters missing in this comment

**Pull request merged**